frontend/coprs_frontend/coprs/views/groups_ns/groups_general.py

```diff
@@ -70,6 +71,11 @@ def list_projects_by_group(group_name, page=1):
     pinned = [pin.copr for pin in PinnedCoprsLogic.get_by_group_id(group.id)] if page == 1 else []
     query = CoprsLogic.get_multiple_by_group_id(group.id)
     query = CoprsLogic.filter_without_ids(query, [copr.id for copr in pinned])
+    query = query.order_by(
+        sqlalchemy.nullsfirst(models.Copr.delete_after),
```

Comment:
do we have index on the delete_after field?

Reply:
We don't. I tried to add it, and alternatively, I tried to add an index on a combination of both `delete_after` and `id`. Neither of them didn't seem to bring any performance improvement. For the record, the code was:

```python
def upgrade():
    op.create_index('copr_delete_after_id', 'copr', ['delete_after', 'id'],
                    unique=False)
```

and

```python
def upgrade():
    op.create_index('copr_delete_after_id', 'copr', ['delete_after'],
                    unique=False)
```

Possibly an error on my side? If not, do you want to add the index anyway?

Reply:
Index with predicate could help.. https://stackoverflow.com/questions/31966218/postgresql-create-an-index-to-quickly-distinguish-null-from-non-null-values ?

Reply:
I haven't found a good index either. Seems like we work with a relatively small number of rows (coprs owned by a single user), yet the query is a relatively complicated set of joins. The production DB performs very well, too. +1